apps/notepad/src/app/component/notes-list/notes-list.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { title } from 'process';
+import { NotesService } from '../../notes.service';
+import { Note } from './note.model';
+
+@Component({
+  selector: 'np-notepad-input',
+  templateUrl: './notes-list.component.html',
+  styleUrls: ['./notes-list.component.scss'],
+})
+export class NotesListComponent implements OnInit {
+  notesList: Note[] = [];
+  searchArray: Note[] = [];
+  searchForm: FormGroup = new FormGroup({
+    searchList: new FormControl(null),
+  });
+
+  constructor(private notesService: NotesService) {}
+  ngOnInit() {
+    this.notesList = this.notesService.fetchNotes();
+    this.searchArray = [...this.notesList];
+  }
+
+  onRemove(noteIndex: number) {
+    // alert(noteIndex + 'Removed');
+    this.notesList.splice(noteIndex, 1);
+    localStorage.setItem('storelist', JSON.stringify(this.notesList));
+  }
+  onSearchSubmit() {
+    const searchTerm = this.searchForm.value.searchList;
```

Comment:
recheck this logic.
for search/filter functionality in ngfor, you can use Pipes
https://coderjony.com/blogs/creating-a-pipe-to-filter-the-items-in-the-list-using-angular-7